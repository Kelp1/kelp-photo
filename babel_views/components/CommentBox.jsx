'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _jsx = function () { var REACT_ELEMENT_TYPE = typeof Symbol === "function" && Symbol.for && Symbol.for("react.element") || 0xeac7; return function createRawReactElement(type, props, key, children) { var defaultProps = type && type.defaultProps; var childrenLength = arguments.length - 3; if (!props && childrenLength !== 0) { props = {}; } if (props && defaultProps) { for (var propName in defaultProps) { if (props[propName] === void 0) { props[propName] = defaultProps[propName]; } } } else if (!props) { props = defaultProps || {}; } if (childrenLength === 1) { props.children = children; } else if (childrenLength > 1) { var childArray = Array(childrenLength); for (var i = 0; i < childrenLength; i++) { childArray[i] = arguments[i + 3]; } props.children = childArray; } return { $$typeof: REACT_ELEMENT_TYPE, type: type, key: key === undefined ? null : '' + key, ref: null, props: props, _owner: null }; }; }();

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var _ref = _jsx('div', {}, void 0, _jsx('img', {
  className: 'comment-userprofile',
  src: 'https://s3-us-west-1.amazonaws.com/welp-icons/userProfile.png',
  height: '50',
  width: '50'
}));

var _ref2 = _jsx('img', {
  className: 'comment-follwers',
  src: 'https://s3-us-west-1.amazonaws.com/welp-icons/followers.png',
  height: '20',
  width: '25'
});

var _ref3 = _jsx('img', {
  className: 'comment-reviews',
  src: 'https://s3-us-west-1.amazonaws.com/welp-icons/reviews.png',
  height: '20',
  width: '25'
});

var _ref4 = _jsx('span', {
  className: 'comment-elite'
}, void 0, '   Elite \'18 ');

var _ref5 = _jsx('div', {
  className: 'comment-sentence'
}, void 0, 'This is a comment');

var _ref6 = _jsx('div', {
  className: 'comment-question'
}, void 0, 'Was this photo ...?');

var _ref7 = _jsx('div', {
  className: 'comment-helpful'
}, void 0, _jsx('button', {}, void 0, ' Helpful  '), _jsx('button', {}, void 0, ' Not Helpful '));

var CommentBox = function (_React$Component) {
  _inherits(CommentBox, _React$Component);

  function CommentBox() {
    _classCallCheck(this, CommentBox);

    return _possibleConstructorReturn(this, (CommentBox.__proto__ || Object.getPrototypeOf(CommentBox)).apply(this, arguments));
  }

  _createClass(CommentBox, [{
    key: 'render',
    value: function render() {
      return _jsx('div', {
        className: 'comment'
      }, void 0, _jsx('div', {
        className: 'commentblock1'
      }, void 0, _ref, _jsx('div', {
        className: 'commentblock2'
      }, void 0, _jsx('div', {}, void 0, _jsx('span', {
        className: 'comment-username'
      }, void 0, this.props.photo.userName)), _jsx('div', {}, void 0, _ref2, _jsx('span', {}, void 0, this.props.photo.userFollowers), _ref3, _jsx('span', {}, void 0, this.props.photo.userReviews, '   '), _ref4))), _ref5, _jsx('div', {
        className: 'comment-date'
      }, void 0, (0, _moment2.default)('' + this.props.photo.date).format('LL')), _ref6, _ref7);
    }
  }]);

  return CommentBox;
}(_react2.default.Component);

exports.default = CommentBox;