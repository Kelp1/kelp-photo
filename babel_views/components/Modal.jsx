'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _jsx = function () { var REACT_ELEMENT_TYPE = typeof Symbol === "function" && Symbol.for && Symbol.for("react.element") || 0xeac7; return function createRawReactElement(type, props, key, children) { var defaultProps = type && type.defaultProps; var childrenLength = arguments.length - 3; if (!props && childrenLength !== 0) { props = {}; } if (props && defaultProps) { for (var propName in defaultProps) { if (props[propName] === void 0) { props[propName] = defaultProps[propName]; } } } else if (!props) { props = defaultProps || {}; } if (childrenLength === 1) { props.children = children; } else if (childrenLength > 1) { var childArray = Array(childrenLength); for (var i = 0; i < childrenLength; i++) { childArray[i] = arguments[i + 3]; } props.children = childArray; } return { $$typeof: REACT_ELEMENT_TYPE, type: type, key: key === undefined ? null : '' + key, ref: null, props: props, _owner: null }; }; }();

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _CommentBox = require('./CommentBox.jsx');

var _CommentBox2 = _interopRequireDefault(_CommentBox);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var _ref = _jsx('span', {
  className: 'modal-underbar-left'
}, void 0, '   Browse All ');

var _ref2 = _jsx('span', {
  className: 'modal-underbar-right'
}, void 0, ' Share   ');

var Modal = function (_React$Component) {
  _inherits(Modal, _React$Component);

  function Modal() {
    _classCallCheck(this, Modal);

    return _possibleConstructorReturn(this, (Modal.__proto__ || Object.getPrototypeOf(Modal)).apply(this, arguments));
  }

  _createClass(Modal, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      return _jsx('div', {
        className: 'modal',
        onClick: function onClick() {
          return _this2.props.handleClose();
        }
      }, void 0, _jsx('span', {
        className: 'close cursor',
        onClick: function onClick() {
          return _this2.props.handleClose();
        }
      }, void 0, 'Close  \xD7'), _jsx('div', {
        className: 'modal-photo',
        onClick: function onClick(event) {
          return event.stopPropagation();
        }
      }, void 0, _jsx('a', {
        className: 'modal-prev',
        onClick: function onClick() {
          return _this2.props.previousPhoto(_this2.props.selectedPhoto);
        }
      }), _jsx('img', {
        className: 'modal-selectedphoto',
        src: this.props.selectedPhoto.photoUrl
      }), _jsx('div', {
        className: 'modal-underbar',
        onClick: function onClick(event) {
          return event.stopPropagation();
        }
      }, void 0, _ref, _jsx('span', {
        className: 'modal-underbar-middle'
      }, void 0, ' ', this.props.modalIndex + 1, ' of ', this.props.totalNumberOfPics), _ref2), _jsx('a', {
        className: 'modal-next',
        onClick: function onClick() {
          return _this2.props.nextPhoto(_this2.props.selectedPhoto);
        }
      })), _jsx('div', {
        className: 'modal-comment',
        onClick: function onClick(event) {
          return event.stopPropagation();
        }
      }, void 0, _jsx(_CommentBox2.default, {
        photo: this.props.selectedPhoto
      })));
    }
  }]);

  return Modal;
}(_react2.default.Component);

exports.default = Modal;