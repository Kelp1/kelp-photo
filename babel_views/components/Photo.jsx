'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _jsx = function () { var REACT_ELEMENT_TYPE = typeof Symbol === "function" && Symbol.for && Symbol.for("react.element") || 0xeac7; return function createRawReactElement(type, props, key, children) { var defaultProps = type && type.defaultProps; var childrenLength = arguments.length - 3; if (!props && childrenLength !== 0) { props = {}; } if (props && defaultProps) { for (var propName in defaultProps) { if (props[propName] === void 0) { props[propName] = defaultProps[propName]; } } } else if (!props) { props = defaultProps || {}; } if (childrenLength === 1) { props.children = children; } else if (childrenLength > 1) { var childArray = Array(childrenLength); for (var i = 0; i < childrenLength; i++) { childArray[i] = arguments[i + 3]; } props.children = childArray; } return { $$typeof: REACT_ELEMENT_TYPE, type: type, key: key === undefined ? null : '' + key, ref: null, props: props, _owner: null }; }; }();

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _CommentBox = require('./CommentBox.jsx');

var _CommentBox2 = _interopRequireDefault(_CommentBox);

var _DescriptionBar = require('./DescriptionBar.jsx');

var _DescriptionBar2 = _interopRequireDefault(_DescriptionBar);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Photo = function (_React$Component) {
  _inherits(Photo, _React$Component);

  function Photo(props) {
    _classCallCheck(this, Photo);

    var _this = _possibleConstructorReturn(this, (Photo.__proto__ || Object.getPrototypeOf(Photo)).call(this, props));

    _this.state = {
      hovered: false
    };
    return _this;
  }

  _createClass(Photo, [{
    key: 'mouseOnPic',
    value: function mouseOnPic() {
      this.setState({
        hovered: true
      });
    }
  }, {
    key: 'mouseLeavePic',
    value: function mouseLeavePic() {
      this.setState({
        hovered: false
      });
    }
  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      if (this.state.hovered === true) {
        return _jsx('span', {
          className: 'picture',
          onMouseEnter: function onMouseEnter() {
            return _this2.mouseOnPic();
          },
          onMouseLeave: function onMouseLeave() {
            return _this2.mouseLeavePic();
          },
          onClick: function onClick() {
            return _this2.props.handleOpen(_this2.props.photo);
          }
        }, void 0, _jsx('img', {
          height: '250',
          width: '250',
          src: this.props.photo.photoUrl,
          className: 'image'
        }), _jsx(_DescriptionBar2.default, {
          photo: this.props.photo
        }));
      } else {
        return _jsx('span', {
          className: 'picture',
          onMouseEnter: function onMouseEnter() {
            return _this2.mouseOnPic();
          },
          onMouseLeave: function onMouseLeave() {
            return _this2.mouseLeavePic();
          },
          onClick: function onClick() {
            return _this2.props.handleOpen(_this2.props.photo);
          }
        }, void 0, _jsx('img', {
          height: '220',
          width: '220',
          src: this.props.photo.photoUrl,
          className: 'image'
        }));
      }
    }
  }]);

  return Photo;
}(_react2.default.Component);

exports.default = Photo;