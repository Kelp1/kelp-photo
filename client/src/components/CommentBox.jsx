import React from 'react';


export default class CommentBox extends React.Component {
  render() {
    return (
      <div>
        <div>{this.props.photo.userName}</div> 
        <div>
          <span>Followers: {this.props.photo.userFollowers}  </span>
          <span>  Reviews: {this.props.photo.userReviews}</span>
        </div> 
        <div>{this.props.photo.date}</div>
      </div>
    )
  }
}