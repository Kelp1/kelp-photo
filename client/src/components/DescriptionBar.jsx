import React from 'react';


export default class DescriptonBar extends React.Component {
  render() {
    return (
      <div className = "descriptionbar">
        <div>
        <img className="comment-userprofile" src="https://s3-us-west-1.amazonaws.com/welp-icons/userProfile.png" height="46" width="46" text-align = "center" /> 
        </div>
        <div className = 'descriptionbarcomment'>
          <div className = "shortcomment"> This is a short comment </div>
          <div className = "descriptionuser"> by {this.props.photo.userName} </div>
        </div>
      </div>
    )
  }
}