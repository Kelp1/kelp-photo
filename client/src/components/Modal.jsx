import React from 'react';

class Modal extends React.Component {
  render() {
    return (
      <div className="modal" onClick={this.props.handleClose} >
        <div className="modal-content">
          <img src={this.props.selectedPhoto.photoUrl} />
        </div>
      </div>
    )
  }
}

export default Modal;