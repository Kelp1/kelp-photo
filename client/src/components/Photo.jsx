import React from 'react';


export default class Photo extends React.Component {
    constructor(props){
        super(props)
        this.state = {
            hovered: false
        }
    }

    mouseOnPic(){
        this.setState({
            hovered: true
        })
    }

    mousLeavePic(){
        this.setState({
            hovered:false
        })
    }

    render() {
        if (this.state.hovered === true) {
            return (
            <span class = "square" onMouseEnter={() => this.mouseOnPic()} onMouseLeave={() => this.mouseLeavePic()}>
                <a href={this.props.url} data-lightbox = "food" ><img height="250" width = "250" src={`${this.props.url}`} className="image" /></a>
            </span>
            ) 
        } else {
            return (
                <span class = "square" onMouseEnter={() => this.mouseOnPic()} onMouseLeave={() => this.mousLeavePic()}>
                    <a href={this.props.url} data-lightbox="food" ><img height="220" width = "220" src={`${this.props.url}`} className="image" /></a>
                </span>
            )
        }
    }
}


