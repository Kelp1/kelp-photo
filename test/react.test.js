import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import axios from 'axios';
import CommentBox from '../client/src/components/CommentBox.jsx';
import DescriptionBar from '../client/src/components/DescriptionBar.jsx';
import Modal from '../client/src/components/Modal.jsx';
import Photo from '../client/src/components/Photo.jsx';
import PhotoGallery from '../client/src/components/PhotoGallery.jsx';

const testData = [
  { "photoId": 4, "photoUrl": "https://s3-us-west-1.amazonaws.com/welp-photos/002-01.jpg", "id": 2, "date": "5/14/2014", "userId": "8498943434", "userName": "gscoular3", "userFollowers": 357, "userReviews": 739 },
  { "photoId": 5, "photoUrl": "https://s3-us-west-1.amazonaws.com/welp-photos/002-02.jpg", "id": 2, "date": "3/23/2015", "userId": "3825079139", "userName": "aivashintsov4", "userFollowers": 867, "userReviews": 890 },
  { "photoId": 6, "photoUrl": "https://s3-us-west-1.amazonaws.com/welp-photos/002-03.jpg", "id": 2, "date": "6/2/2015", "userId": "8788200485", "userName": "glambotin5", "userFollowers": 889, "userReviews": 649 },
  { "photoId": 7, "photoUrl": "https://s3-us-west-1.amazonaws.com/welp-photos/002-04.jpg", "id": 2, "date": "4/7/2017", "userId": "2608088597", "userName": "jferencowicz6", "userFollowers": 607, "userReviews": 68 },
  { "photoId": 8, "photoUrl": "https://s3-us-west-1.amazonaws.com/welp-photos/002-05.jpg", "id": 2, "date": "11/18/2016", "userId": "2773851188", "userName": "lkinneally7", "userFollowers": 571, "userReviews": 649 },
]

const photoGallery = shallow(<PhotoGallery />);
const wrapper = photoGallery

describe('PhotoGallery', () => {

  it('PhotoGallery component should render the wrapper div', () => {
    expect(photoGallery.is('.wrapper')).toBe(true);
  });

  it('should mount only 1', function () {
    expect(mount(<PhotoGallery />).find('.wrapper').length).toBe(1);
  });

  const wrapper = photoGallery;
  wrapper.state().showModal = false;
  it('handleClose in photoGallery toggles the boolean in state.showModal', () => {

    expect(wrapper.state().showModal).toEqual(false);
    wrapper.instance().handleOpen();
    expect(wrapper.state().showModal).toEqual(true);
    wrapper.instance().handleClose();
  });
});


describe('Photo', () => {

  const photo = shallow(<Photo key={4} photo={testData[0]} handleOpen={() => wrapper.setState({ showModal: true })} totalNumberOfPics={5}/>);

  it('Photo component should render the picture', () => {
    expect(photo.is('.picture')).toBe(true);
  });

  it('should mount only 1', function () {
    expect(mount(<Photo key={4} photo={testData[0]} handleOpen={() => wrapper.setState({ showModal: true })} totalNumberOfPics={5} />).find('.picture').length).toBe(1);
  });

  photo.state().hovered = false;
  it('handleClose in photoGallery toggles the boolean in state.showModal', () => {

    expect(photo.state().hovered).toEqual(false);
    photo.instance().mouseOnPic();
    expect(photo.state().hovered).toEqual(true);
    photo.instance().mouseLeavePic();
  });
});

describe('Photo component renders the photo correctly', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(
      <Photo key={4} photo={testData[0]} handleOpen={() => wrapper.setState({ showModal: true })} totalNumberOfPics={5}/>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});

describe('DescriptionBar component renders the description correctly', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(
      <DescriptionBar  photo={testData[0]} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});

describe('CommentBox component renders the comment correctly', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(
      <CommentBox photo={testData[0]} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});

describe('Modal component renders the modal correctly', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(
      <Modal 
        selectedPhoto={testData[1]}
        handleOpen={() => wrapper.setState({ showModal: false })}
        previousPhoto={testData[0]}
        nextPhoto={testData[2]}
        modalIndex={1}
        totalNumberOfPics={5} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});

describe('PhotoGallery component renders the PhotoGallery correctly', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(
      <PhotoGallery/>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});

// // describe('TourPage', () => {

// //   it('should match snapshot', () => {
// //     const images = ['tour', 'test'];
// //     const descriptions = ['descriptions', 'test'];
// //     const view = renderer.create(
// //       <TourPage images={images} descriptions={descriptions} />
// //     ).toJSON();

// //     expect(view).toMatchSnapshot();
// //   });

// // });


// //   it('handleListViewClick in HomePage toggles the boolean in state.listView', () => {

// //     expect(wrapper.state().listView).toEqual(false);
// //     wrapper.instance().handleListViewClick();
// //     expect(wrapper.state('listView')).toEqual(true);
// //     wrapper.instance().handleListViewClick();

// //   });

// //   const tourButton = shallow(<Tour handlePageViewClick={() => wrapper.setState({ tourView: true })} />);

// //   it('handlePageViewClick in Tour toggles the boolean in state.tourView', () => {

// //     expect(wrapper.state().tourView).toEqual(false);
// //     tourButton.find('.tour_button').simulate('click');
// //     expect(wrapper.state('tourView')).toEqual(true);
// //     tourButton.find('.tour_button').simulate('click');

// //   });

// //   const tourButtons = shallow(<TourPage handleListViewClick={() => wrapper.setState({ listView: true })} handlePageViewClick={() => wrapper.setState({ tourView: false })} images={testData.images} descriptions={testData.descriptions} />);

// //   it('handlePageViewClick in TourPage toggles the boolean in state.tourView', () => {

// //     expect(wrapper.state().tourView).toEqual(true);
// //     tourButtons.find('.back_button').simulate('click');
// //     expect(wrapper.state('tourView')).toEqual(false);
// //     tourButtons.find('.back_button').simulate('click');

// //   });

// //   it('handleListViewClick in TourPage toggles the boolean in state.listView', () => {

// //     expect(wrapper.state().listView).toEqual(false);
// //     tourButtons.find('.tour_button').simulate('click');
// //     expect(wrapper.state('listView')).toEqual(true);
// //     tourButtons.find('.tour_button').simulate('click');

// //   });

// //   jest.mock('axios', () => {
// //     const sampleHome = [
// //       { title: 'test home', url: 'sample_url' }
// //     ];

// //     return {
// //       get: jest.fn(() => Promise.resolve(sampleHome)),
// //     };
// //   });

// //   it('get home on #componentDidMount', () => {
// //     const home = homePage;

// //     home.update();
// //     setTimeout(() => {
// //       expect(axios.get).toHaveBeenCalled();
// //       expect(axios.get).toHaveBeenCalledWith('sample_url');
// //       expect(app.state()).toHaveProperty('home', [
// //         { title: 'test home', url: 'sample_url' }
// //       ]);
// //     }, 1000);
// //   });
// // });

// // describe('Introduction', () => {

// //   const home = {
// //     location: 'test',
// //     rating: 3,
// //     title: 'test'
// //   };

// //   it('Introduction component should render the introduction div', () => {
// //     expect(shallow(<Introduction home={home} />).is('.introduction')).toBe(true);
// //   });

// //   it('should mount only 1', function () {
// //     expect(mount(<Introduction home={home} />).find('.introduction').length).toBe(1);
// //   });

// // });


// // describe('Picture', () => {

// //   it('should match snapshot', () => {
// //     const home = {
// //       image: {
// //         living_room: 'test'
// //       }
// //     };
// //     const view = renderer.create(
// //       <Picture home={home} />
// //     ).toJSON();

// //     expect(view).toMatchSnapshot();
// //   });

// // });

// // describe('Save', () => {

// //   it('should match snapshot', () => {
// //     const home = {
// //       saved: true
// //     };
// //     const view = renderer.create(
// //       <Save home={home} />
// //     ).toJSON();

// //     expect(view).toMatchSnapshot();
// //   });

// // });

// // describe('Share', () => {

// //   const home = {
// //     shared: false
// //   };

// //   it('should match snapshot', () => {
// //     const view = renderer.create(
// //       <Share home={home} />
// //     ).toJSON();

// //     expect(view).toMatchSnapshot();
// //   });


// // });

// // describe('Tour', () => {

// //   it('should match snapshot', () => {
// //     const view = renderer.create(
// //       <Tour />
// //     ).toJSON();

// //     expect(view).toMatchSnapshot();
// //   });

// // });

// // describe('TourPage', () => {

// //   it('should match snapshot', () => {
// //     const images = ['tour', 'test'];
// //     const descriptions = ['descriptions', 'test'];
// //     const view = renderer.create(
// //       <TourPage images={images} descriptions={descriptions} />
// //     ).toJSON();

// //     expect(view).toMatchSnapshot();
// //   });

// // });

// // describe('ListView', () => {

// //   it('should match snapshot', () => {
// //     const images = ['list', 'test'];
// //     const view = renderer.create(
// //       <ListView images={images} />
// //     ).toJSON();

// //     expect(view).toMatchSnapshot();
// //   });

// // });