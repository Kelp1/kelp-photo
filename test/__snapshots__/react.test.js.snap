// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentBox component renders the comment correctly renders correctly 1`] = `
<div
  className="comment"
>
  <div
    className="commentblock1"
  >
    <div>
      <img
        className="comment-userprofile"
        height="50"
        src="https://s3-us-west-1.amazonaws.com/welp-icons/userProfile.png"
        width="50"
      />
    </div>
    <div
      className="commentblock2"
    >
      <div>
        <span
          className="comment-username"
        >
          gscoular3
        </span>
      </div>
      <div>
        <img
          className="comment-follwers"
          height="20"
          src="https://s3-us-west-1.amazonaws.com/welp-icons/followers.png"
          width="25"
        />
        <span>
          357
        </span>
        <img
          className="comment-reviews"
          height="20"
          src="https://s3-us-west-1.amazonaws.com/welp-icons/reviews.png"
          width="25"
        />
        <span>
          739
             
        </span>
        <span
          className="comment-elite"
        >
             Elite '18 
        </span>
      </div>
    </div>
  </div>
  <div
    className="comment-sentence"
  >
    This is a comment
  </div>
  <div
    className="comment-date"
  >
    May 14, 2014
  </div>
  <div
    className="comment-question"
  >
    Was this photo ...?
  </div>
  <div
    className="comment-helpful"
  >
    <button>
       Helpful  
    </button>
    <button>
       Not Helpful 
    </button>
  </div>
</div>
`;

exports[`DescriptionBar component renders the description correctly renders correctly 1`] = `
<div
  className="descriptionbar"
>
  <div>
    <img
      className="comment-userprofile"
      height="46"
      src="https://s3-us-west-1.amazonaws.com/welp-icons/userProfile.png"
      width="46"
    />
  </div>
  <div
    className="descriptionbarcomment"
  >
    <div
      className="shortcomment"
    >
       This is a short comment 
    </div>
    <div
      className="descriptionuser"
    >
       by 
      gscoular3
       
    </div>
  </div>
</div>
`;

exports[`Modal component renders the modal correctly renders correctly 1`] = `
<div
  className="modal"
  onClick={[Function]}
>
  <span
    className="close cursor"
    onClick={[Function]}
  >
    Close  Ã—
  </span>
  <div
    className="modal-photo"
    onClick={[Function]}
  >
    <a
      className="modal-prev"
      onClick={[Function]}
    />
    <img
      className="modal-selectedphoto"
      src="https://s3-us-west-1.amazonaws.com/welp-photos/002-02.jpg"
    />
    <div
      className="modal-underbar"
      onClick={[Function]}
    >
      <span
        className="modal-underbar-left"
      >
           Browse All 
      </span>
      <span
        className="modal-underbar-middle"
      >
         
        2
         of 
        5
      </span>
      <span
        className="modal-underbar-right"
      >
         Share   
      </span>
    </div>
    <a
      className="modal-next"
      onClick={[Function]}
    />
  </div>
  <div
    className="modal-comment"
    onClick={[Function]}
  >
    <div
      className="comment"
    >
      <div
        className="commentblock1"
      >
        <div>
          <img
            className="comment-userprofile"
            height="50"
            src="https://s3-us-west-1.amazonaws.com/welp-icons/userProfile.png"
            width="50"
          />
        </div>
        <div
          className="commentblock2"
        >
          <div>
            <span
              className="comment-username"
            >
              aivashintsov4
            </span>
          </div>
          <div>
            <img
              className="comment-follwers"
              height="20"
              src="https://s3-us-west-1.amazonaws.com/welp-icons/followers.png"
              width="25"
            />
            <span>
              867
            </span>
            <img
              className="comment-reviews"
              height="20"
              src="https://s3-us-west-1.amazonaws.com/welp-icons/reviews.png"
              width="25"
            />
            <span>
              890
                 
            </span>
            <span
              className="comment-elite"
            >
                 Elite '18 
            </span>
          </div>
        </div>
      </div>
      <div
        className="comment-sentence"
      >
        This is a comment
      </div>
      <div
        className="comment-date"
      >
        March 23, 2015
      </div>
      <div
        className="comment-question"
      >
        Was this photo ...?
      </div>
      <div
        className="comment-helpful"
      >
        <button>
           Helpful  
        </button>
        <button>
           Not Helpful 
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Photo component renders the photo correctly renders correctly 1`] = `
<span
  className="picture"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    className="image"
    height="220"
    src="https://s3-us-west-1.amazonaws.com/welp-photos/002-01.jpg"
    width="220"
  />
</span>
`;

exports[`PhotoGallery component renders the PhotoGallery correctly renders correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="intro"
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <span>
      <img
        className="emptypage"
        height="220"
        src="https://s3-us-west-1.amazonaws.com/welp-icons/0Pictures.png"
        width="660"
      />
    </span>
  </div>
</div>
`;
